frontend/src/components/Candidate-Decider/CandidateDecider.tsx

```diff
@@ -109,17 +98,64 @@ const CandidateDecider: React.FC<CandidateDeciderProps> = ({ uuid }) => {
     }
   };
 
+  const handleCandidateChange = (candidate: number) => {
+    if (!isSaved) {
+      setIsModalOpen(true);
+    } else {
+      setCurrentCandidate(candidate);
+      populateReviewForCandidate(candidate);
+    }
+  };
+
   return instance.candidates.length === 0 ? (
     <div></div>
   ) : (
     <div className={styles.candidateDeciderContainer}>
+      <Modal open={isModalOpen} onClose={() => setIsModalOpen(false)} size="small">
+        <Modal.Header>Don't Forget To Save!</Modal.Header>
+        <Modal.Content>
+          <p>You have unsaved changes. Do you want to save them before navigating?</p>
+        </Modal.Content>
+        <Modal.Actions>
+          <Button onClick={() => setIsModalOpen(false)}>Cancel</Button>
+          <Button
+            onClick={() => {
+              handleRatingAndCommentChange(
+                currentCandidate,
+                currentRating ?? 0,
+                currentComment ?? ''
+              );
+              if (nextCandidate !== null) {
```

Comment:
How about a function `navigateToNextCandidate` for this and the code that's being duplicated below?